Cynosura.Studio.Generator/PackageFeed/NugetFeed.cs

```diff
@@ -39,37 +41,54 @@ private HttpClient GetHttpClient()
             return httpClient;
         }
 
-        private async Task<string> GetResourceAsync(string type)
+        private async Task<FeedData> GetFeedDataAsync()
         {
             var httpClient = GetHttpClient();
             var feedResult = await httpClient.GetStringAsync(_settings.FeedUrl);
             var feed = feedResult.DeserializeFromJson<FeedData>();
-            var resource = feed.Resources.Where(r => r.Type == type)
-                .Select(r => r.Id)
-                .FirstOrDefault();
-            if (resource != null)
-                resource = Regex.Replace(resource, "/$", "");
-            return resource;
+            return feed;
         }
 
         private async Task<string> GetPackageBaseAddressAsync()
         {
-            return await GetResourceAsync("PackageBaseAddress/3.0.0");
+            var feed = await GetFeedDataAsync();
+            return feed.GetExplicitService("PackageBaseAddress/3.0.0");
         }
 
         private async Task<string> GetSearchAutocompleteServiceAsync()
         {
-            return await GetResourceAsync("SearchAutocompleteService");
+            var feed = await GetFeedDataAsync();
+            return feed.GetExplicitService("SearchAutocompleteService");
+        }
+        
+        private async Task<string> GetRegistrationsBaseUrlAsync()
+        {
+            var feed = await GetFeedDataAsync();
+            return feed.GetType("RegistrationsBaseUrl", _settings.ListingApiVersion ??  "3.6.0");
         }
 
         public async Task<IList<string>> GetVersionsAsync(string packageName)
         {
-            var searchAutocompleteService = await GetSearchAutocompleteServiceAsync();
-            var versionsUrl = $"{searchAutocompleteService}?id={packageName.ToLower()}&prerelease=true";
             var httpClient = GetHttpClient();
-            var versionsResult = await httpClient.GetStringAsync(versionsUrl);
-            var versions = versionsResult.DeserializeFromJson<VersionData>();
-            return OrderVersionsDescending(versions.Data);
+            if (_settings.ListingApi == NugetListingApi.RegistrationsBaseUrl)
+            {
+                var registrationsBase = await GetRegistrationsBaseUrlAsync();
+                var versionsUrl = $"{registrationsBase}/{packageName.ToLower()}";
```

Comment:
It must be $"{registrationsBase}{packageName.ToLower()}/index.json". Then it works for nuget.org too

Reply:
Also need to add
`var handler = new HttpClientHandler
            {
                AutomaticDecompression = DecompressionMethods.GZip
            };
            var httpClient = new HttpClient(handler);`

Reply:
Resolved